core/controllers/topic_editor_test.py

```diff
@@ -348,3 +348,71 @@ def test_assign_topic_manager_role(self):
                 {}, csrf_token=csrf_token, expect_errors=True,
                 expected_status_int=401)
             self.assertEqual(json_response['status_code'], 401)
+
+
+class TopicRightsHandlerTest(BaseTopicEditorControllerTest):
+
+    def test_get_topic_rights(self):
+        """Test the get topic rights functionality.
+        """
+        self.login(self.ADMIN_EMAIL)
+        with self.swap(feconf, 'ENABLE_NEW_STRUCTURES', True):
+            # Test whether admin can access topic rights.
+            json_response = self.get_json(
+                '%s/%s' % (
+                    feconf.TOPIC_RIGHTS_URL_PREFIX, self.topic_id))
+            self.assertEqual(
+                json_response['topic_rights']['topic_id'], self.topic_id)
+            self.assertEqual(
+                json_response['topic_rights']['is_published'], False)
+            self.logout()
+
+
+            self.login(self.NEW_USER_EMAIL)
+            # Test that other users cannot access topic rights.
+            response = self.testapp.get(
+                '%s/%s' % (
+                    feconf.TOPIC_RIGHTS_URL_PREFIX, self.topic_id),
+                expect_errors=True)
+            self.assertEqual(response.status_int, 401)
+            self.logout()
+
+
+class TopicPublishHandlerTest(BaseTopicEditorControllerTest):
+
+    def test_publish_and_unpublish_topic(self):
+        """Test the publish and unpublish functionality.
+        """
```

Comment:
Move to end of previous line. Ditto elsewhere.

Reply:
Done.